impress-org/givewp
# Form Preview step now includes working donation form

Resolves #5008 

## Description

This PR introduces frontend and backend logic required to display a working donation form preview in the Form Preview step of the onboarding wizard. Previously, the step featured a placeholder stand in, rather than an actual working multi-step donation form. This PR replaces the placeholder with an iframe that loads a Multi-Step form. 

_**Note:** This PR does not setup the preview form itself. Instead, it is hardcoded to use a form with the id `18`. To modify which form is displayed (for your own testing) you can change what is returned by the `get_preview_form_id` method in the `Onboarding\Wizard\FormPreview` class._

## Affects

This PR affects frontend logic used to render the donation form preview (DonationForm component), some frontend layout styles, and introduces backend logic to render a form preview at the URL `wp-admin/?page=give-form-preview`.

The PR also includes some minor keyboard accessibility improvements, which came into play as work was done to ensure the the Form Preview step is keyboard accessible (users focus first on the Continue button, then can tab into the form preview, then into onboarding wizard step navigation). 

## Visuals
![FormPreview](https://user-images.githubusercontent.com/5186078/89462574-7a5d0d80-d722-11ea-93bb-adfb04eb79e1.gif)

## Pre-review Checklist

## Testing Instructions
Setup a Multi-Step form, with default settings (and using only the Test gateway). Make sure a featured image has been set for the form. Get the form's ID, then modify the `get_preview_form_id` method in the `Onboarding\Wizard\FormPreview` class to return the desired form ID. 

Navigate to the Onboarding Wizard at `wp-admin/?page=give-onboarding-wizard`, and advance to the Donation Form step. Check that the Wizard is still fully keyboard accessible, and that, on step 4, you can tab into the Form Preview after focus is initially set on the continue button.
